src/Framework/Framework/Compilation/Styles/StyleTreeShufflingVisitor.cs

```diff
@@ -42,12 +42,13 @@ public override void VisitControl(ResolvedControl control)
 
             foreach (var (prop, setter) in control.Properties.ToArray())
             {
-                // remove empty properties
+                // remove empty properties, but preserve empty collections if they are marked as Required
```

Comment:
You mentioned that we want to preserve empty collections. Is it valid also for templates? It seems like something related is failing in unit tests